### DISCUSSION TITLE
Failed to start Persist on ASUS Zenbook

### DISCUSSION THREAD

**Discussion by sergemuur** [tshakalekholoane/bat#73] [General]
Hello there! There are two issues on my quite a new ASUS UM3402YA with LM21.2/Win11 dual boot setup. 

1) Now I have the last 0.14 bat version installed, it works as intended up to the point of rebooting into win11: after that, when I return to LM, it no longer limits the charge for some reason and I have to do "persist" command once again. Kernel and systemd requirements are met:
```
systemd 249 (249.11-0ubuntu3.10)
kernel: 6.2.0-33-generic
```

2) Although it works when not rebooted into win11, it floods the syslog file with a bunch of repeating lines in a cycle, saying it failed to run "persist" command almost every second, so the file grows in size dramatically:

```
Oct  2 16:09:58 serge-Zenbook-UM3402YA-UM3402YA systemd[1]: Starting Persist the battery charging threshold after multi-user...
Oct  2 16:09:58 serge-Zenbook-UM3402YA-UM3402YA systemd[1]: Stopped Persist the battery charging threshold after suspend.
Oct  2 16:09:58 serge-Zenbook-UM3402YA-UM3402YA sh[229726]: /bin/sh: 1: cannot create /sys/class/power_supply/BAT?/charge_control_end_threshold: Directory nonexistent
Oct  2 16:09:58 serge-Zenbook-UM3402YA-UM3402YA systemd[1]: Starting Persist the battery charging threshold after suspend...
Oct  2 16:09:58 serge-Zenbook-UM3402YA-UM3402YA systemd[1]: bat@multi-user.service: Main process exited, code=exited, status=2/INVALIDARGUMENT
Oct  2 16:09:58 serge-Zenbook-UM3402YA-UM3402YA systemd[1]: bat@multi-user.service: Failed with result 'exit-code'.
Oct  2 16:09:58 serge-Zenbook-UM3402YA-UM3402YA systemd[1]: Failed to start Persist the battery charging threshold after multi-user.
Oct  2 16:09:58 serge-Zenbook-UM3402YA-UM3402YA sh[229727]: /bin/sh: 1: cannot create /sys/class/power_supply/BAT?/charge_control_end_threshold: Directory nonexistent
Oct  2 16:09:58 serge-Zenbook-UM3402YA-UM3402YA systemd[1]: bat@suspend.service: Main process exited, code=exited, status=2/INVALIDARGUMENT
Oct  2 16:09:58 serge-Zenbook-UM3402YA-UM3402YA systemd[1]: bat@suspend.service: Failed with result 'exit-code'.
Oct  2 16:09:58 serge-Zenbook-UM3402YA-UM3402YA systemd[1]: Failed to start Persist the battery charging threshold after suspend.
Oct  2 16:09:58 serge-Zenbook-UM3402YA-UM3402YA systemd[1]: bat@multi-user.service: Scheduled restart job, restart counter is at 56267.
Oct  2 16:09:58 serge-Zenbook-UM3402YA-UM3402YA systemd[1]: bat@suspend.service: Scheduled restart job, restart counter is at 40844.
Oct  2 16:09:58 serge-Zenbook-UM3402YA-UM3402YA systemd[1]: Stopped Persist the battery charging threshold after multi-user.
```

There is a _BATT_  folder with _charge_control_end_threshold_  file along the aforementioned path in the 3-d line, but no _BAT?_ directory.

I wanted to uninstall and then reinstall the whole program, but didn't find the way to do it in readme.
I'm a newbie in Linux and git, sorry if I missed sth obvious.

If there's a way to fix that, I could really use your help and guidance!

P.S. I didn't find the step of actual installation in readme, so I used one from a random site with the command:
`sudo install bat /usr/local/bin`
Maybe that might cause some problems?

P.P.S. Ah, yes, one more thing. I ran the suggested command in another discussion concerning Zephrus laptop, it gives me the following:
```
sudo systemctl status bat-boot.service     
Unit bat-boot.service could not be found.
```

**Comment by pepa65** [comment]
That service is now called `bat-multi-user`, so try: `sudo systemctl status bat-multi-user.service`

To uninstall the systemd unit files, do: `sudo bat reset`

To undo the charge threshold, do: `sudo bat threshold 100`

I would recommend cloning the latest version of the repo, and building from that:
```
clone https://github.com/tshakalekholoane/bat
cd bat
make build
sudo install bin/bat /usr/local/bin/
```

That should give you version `0.14`, so I don't know why what you had didn't work.

Could you post the output of `echo /sys/class/power_supply/BAT?`

(The content of the unit files should have `BATT` in your case instead of `BAT?`...)

Did you use `make build` to build?

**Comment by sergemuur** [comment]
The first time I installed the program, I didn't use `make build`, just downloaded the binary and used the `install ` command. 
Now that you suggested building from cloned repo, I used `make build`.

Here's the output of multi-user status:
```
● bat-multi-user.service - Persist the battery charging threshold after multi-user
     Loaded: loaded (/etc/systemd/system/bat-multi-user.service; enabled; vendor preset: enabled)
     Active: active (exited) since Mon 2023-10-02 21:49:49 MSK; 2h 5min ago
    Process: 1566 ExecStart=/usr/bin/sh -c echo 80 > /sys/class/power_supply/BATT/charge_control_end_threshold (code=exited, status=0/SUCCESS)
   Main PID: 1566 (code=exited, status=0/SUCCESS)
        CPU: 2ms

Oct 02 21:49:49 serge-Zenbook-UM3402YA-UM3402YA systemd[1]: Starting Persist the battery charging threshold after multi-user...
Oct 02 21:49:49 serge-Zenbook-UM3402YA-UM3402YA systemd[1]: Finished Persist the battery charging threshold after multi-user.
```

And the echo command's one, as you accurately guessed:

`/sys/class/power_supply/BATT`

I did the reset, cloned the repo, using the commands and built it there. Now Persist works after win11 reboot - that's great!
Unfortunately, syslog still produces the same lines.

Strangely though, `bat -h` gives me:

`2023/10/03 00:19:52 parsing time "" as "2006-01-02": cannot parse "" as "2006"`

and `bat -v` won't show the 0.14 version, giving this:

```
bat 
Copyright (c) 2023 Tshaka Lekholoane.
MIT Licence.
```

Other commands seem to work fine.

**Reply by tshakalekholoane** [reply]
It's possible that it could still be referencing the previous version you had installed because that  information is burned into the binary at build time which is what `make build` does. 

Running `go build` without the required flags or using `go install tshaka.dev/x/bat@latest` will also result in the same errors.

**Reply by sergemuur** [reply]
If that is so, is there a way to do any kind of a clean up to get rid of this old binary which might being referenced?

**Reply by tshakalekholoane** [reply]
Just deleting it from wherever it is should be fine. From your post it seems you have it installed in `/usr/local/bin/` so something like `rm $(which bat)` should be fine.

And then just follow the instructions @pepa65 outlined including running `bat reset` beforehand (note that they seem to have changed since the initial post so make sure you're following the latest change if you're accessing this from your email).

**Reply by sergemuur** [reply]
I did:
```
sudo bat reset
cd /usr/local/bin/
rm -v -i bat
```
There was nothing else in the bin folder in any way connected to bat, so the command just removed the binary. Was it correct?

Thanks for the note about the changed commands!
I deleted the repo manually in the home folder and cloned it again with all the following steps, but these nasty lines in syslog are still appearing(

However, `bat -v` now shows `0.14` version:)

Oddly, syslog prints the errors prior to running persist command and even after deleting the binary in the bin folder and the whole repo folder. 
Seems like there are some extra files deep in the system which are still affected regardless of installing or deleting the program. I wonder, is it possible to find them?

**Reply by tshakalekholoane** [reply]
Yes, that is correct. But you probably ran reset on a newer version so you might have to do it manually. Try running the following commands and then delete the `/etc/systemd/system/bat@.service` file.

```
sudo systemctl disable bat@multi-user.service
sudo systemctl disable bat@suspend.service
```

And just to be thorough, you might also want to delete these as well but it shouldn't be necessary (at least judging from the output you sent).

```
sudo systemctl disable bat@hibernate.service
sudo systemctl disable bat@hybrid-sleep.service
sudo systemctl disable bat@suspend-then-hibernate.service
```

**Reply by sergemuur** [reply]
It finally worked!
My syslog is at peace again:)

Thank you so much, guys, you are doing a truly great job right there. 
Your enthusiasm inspires!

**Reply by tshakalekholoane** [reply]
You're welcome. Thank you for the kinds words!

**Reply by sergemuur** [reply]
Oh, sorry, I've got 1 last basic question: how do I update the program properly when a new release comes out? Do I need to run some `git` command to refresh the repo and then `make build` and `sudo install` or is it done easier?

**Reply by tshakalekholoane** [reply]
Yes, if you're building from source then you have to do all of that manually i.e., `git pull`, `make build` and then `sudo install bin/bat /usr/local/bin/`. 

But now that you mention it, maybe it's something we should include in the Makefile so that it reduces to something like `make install`. What do you think @pepa65?

**Comment by pepa65** [comment]
Yes, `make install` but also `make reinstall` that does `bat reset` first.

**Comment by pepa65** [comment]
> Oh, sorry, I've got 1 last basic question: how do I update the program properly when a new release comes out?

At this point you no longer need any new releases, because you've set the threshold, and you have working systemd unit files to make it persist.

But if you want to upgrade, you could do `bat reset` to make sure no old unit files remain, and then download the newly released binary from https://github.com/tshakalekholoane/bat/releases/download/0.14/bat (the Latest release, in the future the `0.14` will be a higher version) and then make the threshold persistent with `bat persist`.

**Reply by sergemuur** [reply]
Thank you for the suggestion! 
Will keep an eye on your updates just in case)
Good luck!

**Comment by JM-Internet** [comment]
Hi, I upgraded ubuntu to 24.10 and now the bat command is not recognized. Also reinstalling doesn't work and the threshold from the previous ubuntu is still persistent so I cannot charge past 60% which I enabled before the ubuntu upgrade. What can I do to reset/reinstall?

**Comment by sergemuur** [comment]
Hello there, @pepa65 and @tshakalekholoane ! I've been happily using bat since you guys helped me the last time few years ago, but it seems I messed it up once more after a clean arch installation on the same machine (finally ditched Windows entirely). 

ASUS UM3402YA
archlinux 6.16.1-zen1-1-zen

This time I tried downloading the binary from your releases page and using `chmod` and, probably, my stupid imagination.
These are the exact commands I ran:
```
sudo chmod +x /home/sergemuur/Programs/bat
sudo install /home/sergemuur/Programs/bat /usr/local/bin/
```
The program was installed and nothing seemed off (I set the threshold and ran `persist`), except the laptop kept charging past the set threshold. I did the `sudo bat reset` and deleted the initial folder in `/home`, also looked for the services you had pointed out the last time in this discussion, but none of the services was found. 

I reinstalled bat multiple times using good old `git clone` and `make build`, but it never worked. The `/etc/systemd/system/bat@.service` file was properly deleted by the reset command each time.

The `bat-multi-user.service` is active and enabled:
```
● bat-multi-user.service - Persist the battery charging threshold after multi-user
     Loaded: loaded (/etc/systemd/system/bat-multi-user.service; enabled; preset: disabled)
     Active: active (exited) since Sun 2025-08-17 22:26:18 MSK; 3h 23min ago
 Invocation: 7e5c3f1b75c24870ab230bb48403ab11
   Main PID: 978 (code=exited, status=0/SUCCESS)
   Mem peak: 1.7M
        CPU: 9ms

Aug 17 22:26:18 archlinux systemd[1]: Starting Persist the battery charging threshold after multi-user...
Aug 17 22:26:18 archlinux systemd[1]: Finished Persist the battery charging threshold after multi-user.
```
The `/sys/class/power_supply/BATT/charge_control_end_threshold` file has the value of the threshold I set inside. 

The only strangeness is that I haven't been able to find any other services running (like `bat@suspend.service` or `...hibernate` etc.) except for the `bat-multi-user.service`.
```
[sergemuur@archlinux ~]$ sudo systemctl status bat@suspend.service
Unit bat@suspend.service could not be found.
```
Maybe it could be connected with the fact that my system is on btrfs now? I heard that it has a hard time handling hibernation and stuff.

It seems that the answer is very simple but I've run out of any ideas as to how to find it myself. Please, would very much appreciate your help yet one more time with this!

P.S. Oh, one more thing. Every time I reinstall bat, it somehow remembers the threshold I set during previous installation. By default it's not 100. Idk why the `reset` fails to reset it.
